tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -487,15 +518,14 @@ public:
 
    void Finalize()
    {
-      auto resObj = fObjects[0];
-      const auto nSlots = fObjects.size();
-      TList l;
-      l.SetOwner(); // The list will free the memory associated to its elements upon destruction
-      for (unsigned int slot = 1; slot < nSlots; ++slot) {
-         l.Add(fObjects[slot]);
-      }
+      if (fObjects.size() == 1)
+         return;
+
+      Merge(fObjects, /*toselectcorrectoverload=*/0);
```

Comment:
I'm not sure how this works in this case. Does it mean that it will always select the Merge overload that uses TList  (aka what was happening before)? If so, when will the other overload be called in general?

Reply:
that's an excess of care on my part: it's for types that implement _both_ allowed signatures of `Merge`, in which case the second parameter resolves the ambiguity on what to call